src/Nancy.Hosting.Self/NancyHost.cs

```diff
@@ -260,8 +260,8 @@ private Request ConvertRequestToNancyRequest(HttpListenerRequest request)
                 HostName = request.Url.Host,
                 Port = request.Url.IsDefaultPort ? null : (int?)request.Url.Port,
                 BasePath = baseUri.AbsolutePath.TrimEnd('/'),
-                Path = HttpUtility.UrlDecode(relativeUrl),
-                Query = request.Url.Query,
+                Path = relativeUrl,

```

Comment:
Can we just kill the `relativeUrl` variable and do `baseUri.MakeAppLocalPath(request.Url)` directly here? :smile: